Taskfile.yml

```diff
@@ -407,6 +408,33 @@ tasks:
     cmds:
      - "{{.SCRIPTS_ROOT}}/generate-helm-manifest.sh {{.KUBE_RBAC_PROXY}} {{.LOCAL_REGISTRY_CONTROLLER_DOCKER_IMAGE}} {{.PUBLIC_REGISTRY}} {{.LATEST_VERSION_TAG}} `pwd`/{{.CONTROLLER_ROOT}}"
 
+  controller:install-helm:
+    desc: Generate and install helm chart on cluster
+    dir: "{{.CONTROLLER_ROOT}}"
+    cmds:
+    - "helm install --set azureSubscriptionID=$AZURE_SUBSCRIPTION_ID \
+      --set azureTenantID=$AZURE_TENANT_ID \
+      --set azureClientSecret=$AZURE_CLIENT_SECRET \
+      --set azureClientID=$AZURE_CLIENT_ID \
+      --set image.repository={{.LOCAL_REGISTRY_CONTROLLER_DOCKER_IMAGE}}
+      asov2 -n {{.ASO_NAMESPACE}} --create-namespace ./charts/azure-service-operator/"
+    - "kubectl rollout status -n {{.ASO_NAMESPACE}} deployment/azureserviceoperator-controller-manager"
```

Comment:
Does this fail if the rollout fails?
Do we also want to do
`- "kubectl wait --for=condition=ready --timeout=2m pod -n azureserviceoperator-system -l control-plane=controller-manager"` to make sure that things come up?

Reply:
Yes it does fail when the rollout fails. Also we can add wait on this which I missed here. 
However, I think its better to do it the way you suggested by waiting for the pod to be ready.. As deployment "rolled out" is not necessarily without errors.